graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/MessagesAdapterOS2.java

```diff
@@ -253,8 +253,8 @@ private IndexingError.Type errorTypeFromResponse(BulkItemResponse item) {
     private IndexRequest indexRequestFrom(IndexingRequest request) {
         final byte[] body;
         try {
-            body = this.objectMapper.writeValueAsBytes(request.message().toElasticSearchObject(objectMapper, this.invalidTimestampMeter));
-        } catch (JsonProcessingException e) {
+            body = request.message().serialize(SerializationContext.of(objectMapper, invalidTimestampMeter));
```

Comment:
I wonder if we should clear the serialized message from the cache here to avoid keeping it in memory for the lifetime of the message object.

Not sure if we can do that here because the serialized message might be used by other outputs? Right now we only use it in the Elasticsearch/OpenSearch outputs.